### PR TITLE
Add a few configuration-related features

### PR DESCRIPTION
**Pull request by mortenpi**
I figured I'd PR these separately from the documentation for easier review.

### CODE REVIEW

File: test/configurations.jl [hunk 1]
```diff
@@ -50,7 +50,16 @@
 
     @testset "Number of electrons" begin
         @test num_electrons(c"[He]") == 2
+        @test num_electrons(rc"[Xe]") == 54
```
Comment (jagot) [inline test/configurations.jl]: Maybe you meant `@test num_electrons(rc"[Ne]") == 2` here?

File: src/configurations.jl [hunk 2]
```diff
@@ -390,6 +419,25 @@ function Base.:(+)(a::Configuration{O₁}, b::Configuration{O₂}) where {O<:Abs
     Configuration(orbitals, occupancy, states)
 end
 
+"""
+    delete!(c::Configuration, o::AbstractOrbital)
+
+Remove the orbital `o` from the configuration `c`.
```
Comment (jagot) [inline src/configurations.jl]: This is easily confused with the orbital subtraction I implemented recently. Maybe clarify the docstring to say "remove entire subshell corresponding to `o`"?

File: test/configurations.jl (hunk 1 above)
Comment (mortenpi) [reply test/configurations.jl]: Ah, yes, a copy paste mistake.

```suggestion
        @test num_electrons(rc"[He]") == 2
```

File: src/configurations.jl (hunk 2 above)
Comment (mortenpi) [reply src/configurations.jl]:
```suggestion
Remove the entire subshell corresponding to orbital `o` from configuration `c`.
```

**Comment by jagot** [comment]
Thanks!

**Pull request merged**